include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -360,29 +362,34 @@ struct __parallel_transform_reduce_impl
                         auto __group_idx = __item_id.get_group(0);
                         // 1. Initialization (transform part). Fill local memory
                         _Size __n_items;
+
+                        union __storage
+                        {
+                            _Tp __v;
+                            __storage() {}
+                        } __result;
                         if (__is_first)
                         {
-                            __transform_pattern1(__item_id, __n, /*global_offset*/ (_Size)0, __temp_local, __rngs...);
+                            __result.__v = __transform_pattern1(__item_id, __n, /*global_offset*/ (_Size)0, __rngs...);
```

Comment:
A `union `doesn't  call any constructor for type `_Tp`. 
So, `__v` is just a storage, not C++ created object. To be correct with C++ perspective, you should call a move/copy constructor (via a placement new call), like this:
`new (&__in_val.__v) _ValueT(std::move(...));`
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h#L603

After usage of `__v`, you should call a destructor explicitly,  like this: https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h#L619